### PR TITLE
Kafka Incident Agent

### PR DESCRIPTION
**Pull request by gnegri**

### CODE REVIEW

**Comment by yqian10** [comment]
from agent side: 

Screenshot from 2019-05-01 16-50-24

from rabbitMq side: 

Screenshot from 2019-05-01 16-53-00


from backend: 

Screenshot from 2019-05-01 16-54-28


from database: 

Screenshot from 2019-05-01 16-54-54

File: deployment/DeployAgent/roles/install-kafka-incident/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  yum:
+    name: monit
+     state: present
+  when: ansible_pkg_mgr == 'yum'
+- name: Monit | Install Monit
+  apt:
+    name: monit
+    state: present
+  when: ansible_pkg_mgr == 'apt'
+  #- name: Monit | Restart Monit daemon
```
Comment (moharnab123saikia) [inline deployment/DeployAgent/roles/install-kafka-incident/tasks/main.yml]: remove

File: deployment/DeployAgent/roles/install-kafka-incident/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    path: "{{ansible_env.HOME}}/InsightAgent-master"
+- name: Delete ifagent Cron Configuration
+  become: yes
+  file:
+    state: absent
+    path: /etc/cron.d/ifagent
+- name: Copy Agent Code on Machines
+  become: yes
+  copy:
+    src: ../files/insightagent.tar.gz
```
Comment (moharnab123saikia) [inline deployment/DeployAgent/roles/install-kafka-incident/tasks/main.yml]: use the roles folder structure

File: deployment/DeployAgent/roles/install-kafka-incident/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    dest: "{{ansible_env.HOME}}"
+    owner: root
+    group: root
+    mode: 0644
+- name: Install pip
+  become: yes
+  shell: "http_proxy={{ http_proxy }} python {{ansible_env.HOME}}/get-pip.py --no-index --find-links={{ansible_env.HOME}}/pip_packages/"
+- name: Install pip packages for Agent
+  shell: "pip install --no-index --find-links='{{ansible_env.HOME}}/pip_packages/' requests pytz kafka"
+  #- name: Deploy Replay Agent
```
Comment (moharnab123saikia) [inline deployment/DeployAgent/roles/install-kafka-incident/tasks/main.yml]: remove

File: kafka_incident/getincident_kafka.py [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+        to_send_data_dict["projectName"] = agentConfigVars['projectName']
+    else:
+        to_send_data_dict["projectName"] = parameters['project']
+    to_send_data_dict["userName"] = agentConfigVars['userName']
+    to_send_data_dict["instanceName"] = socket.gethostname().partition(".")[0]
+    to_send_data_dict["samplingInterval"] = agentConfigVars['samplingInterval']
+    to_send_data_dict["agentType"] = "LogStreaming"
+
+    to_send_data_json = json.dumps(to_send_data_dict)
+    # logger.debug("TotalData: " + str(len(bytearray(to_send_data_json))))
```
Comment (moharnab123saikia) [inline kafka_incident/getincident_kafka.py]: remove

File: kafka_incident/getincident_kafka.py [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+    parse_consumer_messages(consumer, filter_hosts)
+    consumer.close()
+    logger.info("Closed log consumer number " + consumer_id)
+
+
+if __name__ == "__main__":
+    logger = set_logger_config()
+    parameters = get_parameters()
+    agentConfigVars = get_agent_config_vars()
+    # reportingConfigVars = get_reporting_config_vars()
```
Comment (moharnab123saikia) [inline kafka_incident/getincident_kafka.py]: remove

File: kafka_incident/getincident_kafka.py [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+    consumer.subscribe([topic])
+    parse_consumer_messages(consumer, filter_hosts)
+    consumer.close()
+    logger.info("Closed log consumer number " + consumer_id)
+
+
+if __name__ == "__main__":
+    logger = set_logger_config()
+    parameters = get_parameters()
+    agentConfigVars = get_agent_config_vars()
```
Comment (moharnab123saikia) [inline kafka_incident/getincident_kafka.py]: remove camelcase

File: deployment/DeployAgent/roles/install-kafka-incident/tasks/main.yml (hunk 2 above)
Comment (gnegri) [reply deployment/DeployAgent/roles/install-kafka-incident/tasks/main.yml]: out of scope

**Pull request merged**